http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreterSpec.scala

```diff
@@ -349,27 +430,60 @@ class AKSInterpreterSpec extends AnyFlatSpecLike with TestComponent with Leonard
-
-      createdDatabase <- aksInterp.createWsmDatabaseResource(saveApp,
-                                                             workspaceId,
-                                                             ControlledDatabase("test", false),
-                                                             "wds",
-                                                             Some(owner),
-                                                             mockControlledResourceApi
+
+      controlledDatabases <- aksInterp.createOrFetchWsmDatabaseResources(saveApp,
+                                                                         saveApp.appType,
+                                                                         workspaceIdForAppCreation,
+                                                                         app.appResources.namespace.value,
+                                                                         Option("idworkflows_app"),
+                                                                         lzResources,
+                                                                         mockResourceApi
       )
 
       controlledResources <- appControlledResourceQuery
         .getAllForAppByStatus(appId.id, AppControlledResourceStatus.Created)
         .transaction
     } yield {
-      createdDatabase.getAzureDatabase.getMetadata.getName shouldBe "test_wds"
-      createdDatabase.getAzureDatabase.getAttributes.getDatabaseOwner shouldBe owner
-      controlledResources.size shouldBe 1
-      controlledResources.head.resourceId.value shouldBe createdDatabase.getResourceId
+      controlledDatabases.size shouldBe 2
+      controlledDatabases.head.wsmDatabaseName shouldBe "cbas"
+      controlledDatabases.head.azureDatabaseName shouldBe "cbas_ns"
+      controlledDatabases(1).wsmDatabaseName shouldBe "cromwellmetadata"
+      controlledDatabases(1).azureDatabaseName shouldBe "cromwellmetadata_ns"
+
+      controlledResources.size shouldBe 2
       controlledResources.head.resourceType shouldBe WsmResourceType.AzureDatabase
       controlledResources.head.status shouldBe AppControlledResourceStatus.Created
       controlledResources.head.appId shouldBe appId.id
+      controlledResources(1).resourceType shouldBe WsmResourceType.AzureDatabase
+      controlledResources(1).status shouldBe AppControlledResourceStatus.Created
+      controlledResources(1).appId shouldBe appId.id
+
+      val createDbCaptor = ArgumentCaptor.forClass(classOf[CreateControlledAzureDatabaseRequestBody])
+      verify(mockControlledResourceApi, atLeastOnce()).createAzureDatabase(createDbCaptor.capture(), any())
+      // there are multiple invocations of "createAzureDatabase" in the mock API before this test is run.
+      // We extract the last 2 invocations which reflect the calls made as part of this test.
+      // Call for "cbas" db creation will be second last in the list
+      val cbasDbCreationCallIndex = createDbCaptor.getAllValues.size() - 2
```

Comment:
For reviewers: Is there a better way to extract the last 2 invocations to the `mockControlledResourceApi`? I tried the approach of resetting the mock in `beforeEach` test but ran into some issues including the tests finding the mock to be null even if I setup the mock each time in the `beforeEach` section. Other way would be to run this test first but I feel tests shouldn't have expectations in them that they would be always run first.

Reply:
hmm there probably is, I'm not sure off the top of my head, you might be able to do `createDbCaptor.getAllValues[-1]`? but I'd have to poke around more to confirm